### ISSUE TITLE
能否导出特征对应的embedding向量？

### ISSUE THREAD

**Issue by sunjiaxin111** [aksnzhy/xlearn#176]
我想问下这个库训练完FM或FFM后，能导出特征对应的embedding向量么

**Comment by aksnzhy** [comment]
可以的，详细信息可以参考文档。

**Comment by sunjiaxin111** [comment]
请问这里的setTXTModel方法怎么不管用？

**Comment by aksnzhy** [comment]
@sunjiaxin111 , 您好，方便把代码贴上来吗，我帮你看看。

**Comment by sunjiaxin111** [comment]
import xlearn as xl

# Training task
fm_model = xl.create_fm()  # Use factorization machine
path = '../data/avazu/'
fm_model.setTrain(path + 'small_train_fm.csv')  # Training data

# param:
#  0. Binary classification task
#  1. learning rate: 0.2
#  2. lambda: 0.002
#  3. metric: accuracy
param = {'task': 'binary', 'lr': 0.2, 'lambda': 0.002, 'metric': 'auc'}

# 训练得到模型
fm_model.fit(param, "model.out")

# 预测
fm_model.setSigmoid()
fm_model.setTXTModel('model.txt')
fm_model.setTest(path + 'small_test_fm.csv')
fm_model.predict('model.out', "output.txt")

**Comment by sunjiaxin111** [comment]
‘’
import xlearn as xl

# Training task
fm_model = xl.create_fm()  # Use factorization machine
path = '../data/avazu/'
fm_model.setTrain(path + 'small_train_fm.csv')  # Training data

# param:
#  0. Binary classification task
#  1. learning rate: 0.2
#  2. lambda: 0.002
#  3. metric: accuracy
param = {'task': 'binary', 'lr': 0.2, 'lambda': 0.002, 'metric': 'auc'}

# 训练得到模型
fm_model.fit(param, "model.out")

# 预测
fm_model.setSigmoid()
fm_model.setTXTModel('model.txt')
fm_model.setTest(path + 'small_test_fm.csv')
fm_model.predict('model.out', "output.txt")‘
‘’’

**Comment by sunjiaxin111** [comment]
`import xlearn as xl

# Training task
fm_model = xl.create_fm()  # Use factorization machine
path = '../data/avazu/'
fm_model.setTrain(path + 'small_train_fm.csv')  # Training data

# param:
#  0. Binary classification task
#  1. learning rate: 0.2
#  2. lambda: 0.002
#  3. metric: accuracy
param = {'task': 'binary', 'lr': 0.2, 'lambda': 0.002, 'metric': 'auc'}

# 训练得到模型
fm_model.fit(param, "model.out")

# 预测
fm_model.setSigmoid()
fm_model.setTXTModel('model.txt')
fm_model.setTest(path + 'small_test_fm.csv')
fm_model.predict('model.out', "output.txt")`

**Comment by sunjiaxin111** [comment]
![image](https://user-images.githubusercontent.com/13600388/49053241-5990a500-f22a-11e8-9dad-846604adbdaf.png)

**Comment by aksnzhy** [comment]
@sunjiaxin111 我看了你的代码，setTXTModel() 需要放在 fit API 之前，fit 函数会把 TXT 模型输出到对应路径。

**Comment by sunjiaxin111** [comment]
@aksnzhy 我试试

**Comment by sunjiaxin111** [comment]
@aksnzhy 可以了，不过你们的文档里写的还是容易造成误解的
![image](https://user-images.githubusercontent.com/13600388/49053433-1420a780-f22b-11e8-965a-463d9674eaed.png)

**Comment by aksnzhy** [comment]
@sunjiaxin111 谢谢，我去把这里改一下。

**Comment by better629** [comment]
> 可以的，详细信息可以参考文档。

How ? can't find keyword 'embedding' in your code or doc.
Thx!

**Comment by aksnzhy** [comment]
@better629 Hi, you can see it here: https://xlearn-doc.readthedocs.io/en/latest/cli_api/index.html#model-output
